core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -486,9 +488,12 @@ private[kafka] class Processor(val id: Int,
   private def processCompletedReceives() {
     selector.completedReceives.asScala.foreach { receive =>
       try {
-        val channel = selector.channel(receive.source)
-        val session = RequestChannel.Session(new KafkaPrincipal(KafkaPrincipal.USER_TYPE, channel.principal.getName),
-          channel.socketAddress)
+        val openChannel = selector.channel(receive.source)
+        val session = {
+          // Only methods that are safe to call on a closed channel should be invoked on 'channel'.
```

Comment:
Maybe `closed` should be replaced by `disconnected` since the channel is not really closed (if I have read the code correctly).